pvlib/shading.py

```diff
@@ -553,3 +553,109 @@ def shaded_fraction1d(
     )
 
     return np.clip(t_asterisk, 0, 1)
+
+
+def martinez_shade_factor(shaded_fraction, N_shaded_blocks, N_total_blocks):
+    r"""
+    A shading correction factor for the direct and circumsolar incident
+    irradiance of non-monolithic Silicon
+    modules and arrays with an arbitrary number of bypass diodes.
+
+    .. versionadded:: 0.11.0
+
+    Parameters
+    ----------
+    shaded_fraction : numeric
+        Surface shaded fraction. Unitless.
```

Comment:
It may be helpful to specify that this model wants a 2-D (area) fraction, which is not always equivalent to what other pvlib functions (`pvlib.shading.shaded_fraction1d`) return.